File: src/LinkedList/SinglyLinkedList/InsertEnd/Node.java
package LinkedList.SinglyLinkedList.InsertEnd;

class Node {
    public int data;
    public Node next;

    public Node (int data) {
        this.data = data;
        this.next = null;
    }
}

class SinglyList {
    private Node head;

    public void insertNodeAtEnd(int value) {
        Node newNode = new Node(value);
        if(head == null) {
            newNode.next = head;
            head = newNode;
            return;
        }
        Node current = head;
        while(current.next != null) {
            current = current.next;
        }
        current.next = newNode;
    }

    public void printList() {
        Node cur = head;
        while (cur != null) {
            System.out.print(cur.data + " --> ");
            cur = cur.next;
        }
        System.out.println("null");
    }

    public static void main(String[] args) {
        SinglyList sl = new SinglyList();

        sl.head = new Node(1);
        Node second = new Node(2);
        Node third = new Node(3);
        Node forth = new Node(4);

        sl.head.next = second;
        second.next = third;
        third.next = forth;

        sl.insertNodeAtEnd(5);
        sl.printList();
    }
}

File: linked_list/sinlgly_list/print_list/node.py
class Node:
  def __init__(self, data: int) -> None:
    self.data = data
    self.next = None

class LinkedList:
  def __init__(self) -> None:
    self._head = None

  def display_list(self) -> None:
    cur = self._head

    while cur:
      print(cur.data, end=' --> ')
      cur = cur.next
    print('null')

sl = LinkedList()
sl._head = Node(1)
sl._head.next = Node(2)
sl._head.next.next = Node(3)
sl.display_list()